### ISSUE TITLE
DL3018 with !pyc option

### ISSUE THREAD

**Issue by AKorezin** [hadolint/hadolint#956]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

In *alpine:3.18* `__pycache__` was split into separate packages with `-pyc` suffix.
[Release document](https://www.alpinelinux.org/posts/Alpine-3.18.0-released.html#significant_changes) recommends using `apk add !pyc`.

### Actual behavior

`apk add !pyc` triggers DL3018.
```
DL3018 warning: Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`
```

### Steps to reproduce the behavior

**Output of `hadolint --version`**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile**

```Dockerfile
FROM alpine:3.18

RUN apk add --no-cache !pyc python3=3.11.3-r11
```

**Additional environment details (OS, stack version, etc.)**
